test/mainController.test.ts

```diff
@@ -99,9 +54,32 @@ suite('MainController Tests', async () => {
         // Resetting call back variables
         docUriCallback = '';
         newDocUriCallback = '';
-    });
 
-    mainController = await setupMaincontroller();
+
+        // Using the mainController that was instantiated with the extension
+        mainController = await Extension.getController();
```

Comment:
Why are you refactoring this anyways? Was there an issue with how it was done before?

Reply:
Yeah, the mainController tests fail otherwise

Reply:
And why did they fail? I assume they weren't failing before you made these changes...

Reply:
They were that's why I changed them. There's no other reason to change the tests otherwise.